### ISSUE TITLE
Console error in QSelect's add-unique new-value-mode when value prop is string

### ISSUE THREAD

**Issue by tomers** [quasarframework/quasar#4807]
**Describe the bug**
QSelect's API defines the `model` prop to be of type "Number | String | Array". However, when using QSelect's `new-value-mode="add-unique"` mode, and setting the model to an empty string, we get a console error:
> TypeError: this.innerValue.map is not a function

Codepen based on https://quasar.dev/vue-components/select#The-new-value-mode-prop:
https://codepen.io/tomers-the-encoder/pen/zgPZEO

It is understood that the model in such mode should be either null or an array, but the error should be more descriptive for the end user in such case.

Also, it feels as if there is too much functionality in the QSelect component. 
Instead of having to specify
```
      <q-select
        label="Mode: 'add-unique'"
        filled
        v-model="modelAddUnique"
        use-input
        use-chips
        multiple
        hide-dropdown-icon
        input-debounce="0"
        new-value-mode="add-unique"
        style="width: 250px"
      />
```
in order to create a component that allows specifying unique list of values, as a user I would expect to have a dedicated component for that purpose (possibly base on QSelect).
Maybe the QSelect component can be wrapper in helper components that would allow syntactic sugar for users...
```
      <q-unique-values
        label="Mode: 'add-unique'"
        filled
        v-model="model"
        style="width: 250px"
      />
```

**Comment by pdanpdan** [comment]
So create as many components as you like. Nobody is stopping you. It would be around 10 lines of code.

**Comment by rstoenescu** [comment]
Fix will be available in 1.0.5.
Thx for reporting!

**Comment by tomers** [comment]
> So create as many components as you like. Nobody is stopping you. It would be around 10 lines of code.

My comment and this discussion are a bit off-topic, but anyway I would like to reply.
That is exactly what I did, and I'm moving on with my development tasks. However... I wrote this not for me personally, but for other current and future devs who would encounter this.
I think that the framework could benefit from introducing some helper components, wrapping the core components such as QInput and QSelect, which have so many props and so many permutations of these prop that provide various functionalities. The fact that there are so many props and that the documentation page for these core components is so long, is a bad signal TMHO.
